views/fields/editable_text.tt

```diff
@@ -0,0 +1,28 @@
+[%-
+  # Template for single-value editable input
+  # Implementation Example:
+  #
+  # INCLUDE fields/editable_text.tt
+  #   value = user.name
+  #   label = "Email address"
+  #   placeholder = "name@example.com"
+  #   type = "email"
+-%]
+
+<div class="input-group mb-3">
+    <input 
+        type=" [% type OR 'text' %]"
+        class="form-control"
+        value="[% value %]"
+        placeholder="[% placeholder %]"
+        aria-label="[% label %]"
+        aria-describedby="update-button"
+        name = "[% name %]" />
+    <button
+        type="[% btn_type || 'button' %]"[% IF btn_val %]
+        value="[% btn_val %]"
+        id="[% btn_val %]"
+        name="[% btn_val %]"[% END %]
+        class="btn btn-success"
+        id="update-button">Update</button>
+</div>
```

Comment:
I'm wondering rather than creating a new field type, whether we can just use the same style as other settings pages, with the submit button at the bottom of the page?

Reply:
This is fine, and doable - I created this when it was originally an AJAX request. The only issue I can percieve is that the report list table would then have what looks like a "hanging" submit button in the bottom right which would have an "unclear" purpose in my opinion, so it was clearer in my opinion (as it was only to change a single field) to do it this way. With that in mind for the report view, it was then added in the same manner to the settings view in order to preserve consistency.